### PR TITLE
Content Type Fix

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
First PR in converting to reliable `withX`, `withoutX` and `withXOption` functions.

### CODE REVIEW

File: core/src/main/scala/org/http4s/headers/Content-Type.scala [hunk 1]
```diff
@@ -7,7 +7,7 @@ import org.http4s.util.Writer
 object `Content-Type` extends HeaderKey.Internal[`Content-Type`] with HeaderKey.Singleton {
   def apply(mediaType: MediaType, charset: Charset): `Content-Type` =
     apply(mediaType, Some(charset))
-  implicit def apply(mediaType: MediaType): `Content-Type` = apply(mediaType, None)
+  def apply(mediaType: MediaType): `Content-Type` = apply(mediaType, None)
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/headers/Content-Type.scala]: Removed implicit application

File: core/src/main/scala/org/http4s/MessageOps.scala [hunk 2]
```diff
@@ -39,11 +39,14 @@ trait MessageOps[F[_]] extends Any {
...
-      case None => filterHeaders(_.is(`Content-Type`))
-    }
+  final def withContentType(contentType: `Content-Type`)(implicit F: Functor[F]): Self =
+    putHeaders(contentType)
+
+  final def withoutContentType(implicit F: Functor[F]): Self = filterHeaders(_.is(`Content-Type`))
+
+  final def withContentTypeOption(contentTypeO: Option[`Content-Type`])(
+      implicit F: Functor[F]): Self =
+    contentTypeO.fold(withoutContentType)(withContentType)
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/MessageOps.scala]: This syntax should be easily extended to other options.

Comment (rossabaker) [reply core/src/main/scala/org/http4s/MessageOps.scala]: There are a bunch on the builders, I think.

**Comment by rossabaker** [comment]
I can't approve on phone, but :+1:

**Pull request merged**